taca/cleanup/cleanup.py

```diff
@@ -234,14 +236,41 @@ def cleanup_irma(days_fastq, days_analysis, only_fastq, only_analysis, status_db
 
     #compile list for project to delete
     project_clean_list, project_processed_list = ({}, [])
-    if not list_only:
+    if not list_only and not clean_undetermined:
         logger.info("Building initial project list for removing data..")
     if only_fastq:
         logger.info("Option 'only_fastq' is given, so will not look for analysis data")
     elif only_analysis:
         logger.info("Option 'only_analysis' is given, so will not look for fastq data")
-     
-    if only_analysis:
+    
+    if clean_undetermined:
+        all_undet_files = []
+        for flowcell_dir in flowcell_dir_root:
+            for fc in [d for d in os.listdir(flowcell_dir) if re.match(filesystem.RUN_RE,d)]:
+                fc_abs_path = os.path.join(flowcell_dir, fc)
+                with filesystem.chdir(fc_abs_path):
```

Comment:
Yes @vezzi , the script goes back to original directory when it leaves the `with` statement here. So in this context, it goes inside each flowcell and then collect the undetermined files (with absolute path) 👍